### DISCUSSION TITLE
Is there solutions for random message id?

### DISCUSSION THREAD

**Discussion by ghost** [libp2p/rust-libp2p#4062] [Q&A] [answered]
I am working on p2p search engine, and every search request cannot be unique. In chat example message id is calculated with hash function: 
```rust  
let message_id_fn = |message: &gossipsub::Message| {
        let mut s = DefaultHasher::new();
        message.data.hash(&mut s);
        gossipsub::MessageId::from(s.finish().to_string())
};
```
So, what i should do? Maybe generate random string?

**Comment by dariusc93** [comment, marked as answer]
Assuming I am understanding your question, you can remove the config entry for computing the id so it would use the default which is located https://github.com/libp2p/rust-libp2p/blob/c93f753018af3a63c07e38c09672e0e1e8f1a385/protocols/gossipsub/src/config.rs#L412-L425. You can also use your own logic to compute the id too but default should suffice.

**Reply by ghost** [reply]
Thank, it will be better than other solution(use timestamp)
